lib/ontologies_linked_data/utils/notifier.rb

```diff
@@ -0,0 +1,125 @@
+module LinkedData
+  module Utils
+    class Notifier
+      def self.notify(options = {})
+        return unless LinkedData.settings.enable_notifications
+
+        headers = { 'Content-Type' => 'text/html' }
+        sender = options[:sender] || LinkedData.settings.email_sender
+        recipients = Array(options[:recipients]).uniq
+        raise ArgumentError, 'Recipient needs to be provided in options[:recipients]' if !recipients || recipients.empty?
+
+        # By default we override all recipients to avoid
+        # sending emails from testing environments.
+        # Set `email_disable_override` in production
+        # to send to the actual user.
+        unless LinkedData.settings.email_disable_override
+          headers['Overridden-Sender'] = recipients
+          recipients = LinkedData.settings.email_override
+        end
+
+        Pony.mail({
+                    to: recipients,
+                    from: sender,
+                    subject: options[:subject],
+                    body: options[:body],
+                    headers: headers,
+                    via: :smtp,
+                    enable_starttls_auto: LinkedData.settings.enable_starttls_auto,
+                    via_options: mail_options
+                  })
+      end
+
+      def self.notify_ontoportal_admins_grouped(subject, body)
+        notify_mails_grouped subject, body, ontoportal_admin_emails if LinkedData.settings.enable_administrative_notifications
+      end
+
+      def self.notify_subscribed_separately(subject, body, ontology, notification_type)
+        mails = subscribed_users_mails(ontology, notification_type)
+        notify_mails_separately subject, body, mails
+      end
+
+      def self.notify_administrators_grouped(subject, body, ontology)
+        notify_mails_grouped subject, body, ontology_admin_emails(ontology)
+      end
+
+      def self.notify_mails_separately(subject, body, mails)
+        mails.each do |mail|
+          notify_mails_grouped subject, body, mail
+        end
+      end
+
+      def self.notify_mails_grouped(subject, body, mail)
+        options = {
+          subject: subject,
+          body: body,
+          recipients: mail
+        }
+        notify(options)
+      end
+
+      def self.notify_ontoportal_admins(subject, body)
+        notify_mails_grouped subject, body, ontoportal_admin_emails if LinkedData.settings.enable_administrative_notifications
+      end
+
+      def self.ontology_admin_emails(ontology)
+        ontology.bring :administeredBy if ontology.bring? :administeredBy
+        recipients = []
+        ontology.administeredBy.each do |user|
+          user.bring(:email) if user.bring?(:email)
+          recipients << user.email
+        end
+        recipients
+      end
+
```

Comment:
you can add (and use) the following helper/facets/alias here 

```ruby 
def self.support_mail
 Array(LinkedData.settings.support_contact_email)
end
```
```ruby 
def self.notification_enabled?
   LinkedData.settings.enable_administrative_notifications
end 
```

this will make it easier to read

Reply:
thanks for the suggestion

addressed in d0b6ad881d2d96008547315648b74477918fab4b